python/pfs/drp/stella/lsf.py

```diff
@@ -0,0 +1,688 @@
+        ``y`` should produce an interpolator that can be called with an array of
+        ``x`` values at which to interpolate.
+    """
+    def __init__(self, length, interpolator=DEFAULT_INTERPOLATOR):
+        self.length = length
+        self.interpolator = interpolator
+
+    def computeImage(self, point=None):
+        """Return an image with the LSF inserted at the nominated position
+
+        This method does the same as the ``computeArray`` method, but packages
+        the result as an ``Image``.
+
+        Parameters
+        ----------
+        point : `lsst.geom.Point2D`, optional
+            Position at which to evaluate the LSF.
+
+        Returns
+        -------
+        image : `lsst.afw.image.Image`
+            Image containing the LSF inserted.
+        """
+        if point is None:
+            point = self.getAveragePosition()
+        return ImageF(self.computeArray(point.getX())[np.newaxis, :].astype(np.float32))
+
+    def computeArray(self, center):
+        """Return an array with the LSF inserted at the nominated position
+
+        Besides the difference in return types from the ``computeKernel``
+        method (this returns an array, while ``computeKernel`` returns a
+        `Kernel1D`), this method allows positioning of the LSF at sub-pixel
+        positions. This is the method you want to use if you want a model for
+        a particular line in a spectrum, e.g., to measure the flux and/or
+        subtract the line.
+
+        Parameters
+        ----------
+        center : `float`, pixels
+            Position at which to insert the LSF.
+
+        Returns
+        -------
+        array : `numpy.ndarray`, floating-point
+            Array containing the LSF inserted.
+        """
+        return self.computeKernel(center).toArray(self.length, center, interpolator=self.interpolator)
+
+    def computeKernelImage(self, point=None):
+        """Return an image of the kernel for the nominated position
+
+        This method does the same as the ``computeKernel`` method, but packages
+        the result as an ``Image``.
+
+        Parameters
+        ----------
+        point : `lsst.geom.Point2D`, optional
+            Position at which to evaluate the LSF.
+
+        Returns
+        -------
+        kernelImage : `lsst.afw.image.Image`
+            Image with a model of the LSF at the nominated position.
+        """
+        if point is None:
+            point = self.getAveragePosition()
+        kernel = self.computeKernel(point.getX())
+        return ImageF(kernel.values[np.newaxis, :].astype(np.float32), True, Point2I(-kernel.center, 0))
+
+    @abstractmethod
+    def computeKernel(self, center):
+        """Return a kernel for the nominated position
+
+        Besides the difference in return types from the ``computeArray``
+        method (this returns a `Kernel1D`, while ``computeArray`` returns a
+        an array), this method provides a model for the LSF centered at zero.
```

Comment:
What does, "centred at zero" mean?  Not pixel `0`, but in the sense of your `Kernel1D` class?